### PR TITLE
disable submenus rendered as dropdowns

### LINKED ISSUE #109701: Action icons that are submenus don't handle empty well
Related to #107147

If you have a command in an action bar (say in `editor/title`) that is a submenu with no items (or no visible items), there is no visible indication -- the icon is there, but does nothing if you click on it. I would expect similar behavior to #107147

### PR DESCRIPTION
**Pull request by sbatten**
This PR fixes #109701

In PR to review the behavior:

For regular submenus, do what native has always done and show an empty menu with a single disabled item.
For icon/dropdown submenus, disable the item if there are no entries provided

### CODE REVIEW

File: src/vs/base/browser/ui/dropdown/dropdownActionViewItem.ts [hunk 1]
```diff
@@ -134,6 +139,26 @@ export class DropdownMenuActionViewItem extends BaseActionViewItem {
...
+		} else {
+			if (this.actionItem) {
+				this.actionItem.classList.add('disabled');
+			}
+
+			if (this.element) {
+				this.element.classList.add('disabled');
+			}
+		}
+	}
```
Comment (eamodio) [inline src/vs/base/browser/ui/dropdown/dropdownActionViewItem.ts:161]: Maybe change to something more succinct?
```suggestion
updateEnabled(): void {
	const disabled = !this.getAction().enabled;
	this.actionItem?.classList.toggle('disabled', disabled);
	this.element?.classList.toggle('disabled', disabled);
}
```

**Comment by eamodio** [comment]
LGTM!

File: src/vs/base/browser/ui/dropdown/dropdownActionViewItem.ts [hunk 2]
```diff
@@ -134,6 +139,26 @@ export class DropdownMenuActionViewItem extends BaseActionViewItem {
 			this.dropdownMenu.show();
 		}
 	}
+
+	updateEnabled(): void {
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/dropdown/dropdownActionViewItem.ts:143]: Shouldn't this be `protected`?

**Review by joaomoreno — APPROVED** [review]
I didn't run this, but if it works, it lgtm.

**Pull request merged**